File: Template/JS/StringParent.js
import Parent from "./Parent";

export default class StringParent extends Parent {
  constructor(input) {
    super();
    this.input = input;
  }

  open() {
    super.open();
    console.log("OPEN String Parent\n");
  }

  close() {
    super.close();
    console.log("CLOSE String Parent");
  }

  print() {
    super.print();
    console.log(this.input);
  }
}

File: Template/Java/src/StringParent.java
public class StringParent extends Parent{
    private String words;

    public StringParent(String words){
        this.words = words;
    }

    @Override
    public void open() {
        System.out.println("--- Open String Parent ---");
    }

    @Override
    public void close() {
        System.out.println("--- Close String Parent ---");
    }

    @Override
    public void print() {
        System.out.print("|");
        System.out.print(this.words);
        System.out.println("|");
    }
}
